client/app/lib/redux/modules/payment/creditcard.coffee

```diff
@@ -0,0 +1,72 @@
+_ = require 'lodash'
+immutable = require 'app/util/immutable'
+
+{ makeNamespace, expandActionType,
+  normalize, defineSchema } = require 'app/redux/helper'
+
+withNamespace = makeNamespace 'koding', 'payment', 'creditcard'
+
+REMOVE = expandActionType withNamespace 'REMOVE'
+
+customer = require './customer'
+
+reducer = (state = null, action) ->
+
+  switch action.type
+
+    when customer.LOAD.SUCCESS, customer.CREATE.SUCCESS, customer.UPDATE.SUCCESS
+      normalized = normalize action.result, customer.schema
+      c = normalized.first 'customer'
+
+      if c.default_source
+        return normalized.entities.sources[c.default_source]
+
+      return null
+
+    when REMOVE.SUCCESS, customer.REMOVE.SUCCESS
+      return null
+
+    else
+      return state
```

Comment:
This one is a little bit tricky. Because the removed state itself is `null`. But you're right about `else` statement, i am fixin' it.